MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -70,73 +70,61 @@ public partial class Texture2D : Texture
         private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat format, SurfaceType type, bool shared)
         {
             this.glTarget = TextureTarget.Texture2D;
+            format.GetGLFormat(GraphicsDevice, out glInternalFormat, out glFormat, out glType);
+            // GL texture is generated on first access in PlatformSetData
```

Comment:
@KonajuGames @dellis1972 @cra0zy 

I still feel like at the very least `GenerateGLTextureIfRequired();` should be called here.  Having texture creation not occur until `SetData` (or even `GetData`) is inconsistent with XNA behavior and even how MG works under DirectX and all our console platforms.

For example texture creation can fail...  is it better for this exception from the GL error to come from `SetData` or when `new Texture2D` is called?